### ISSUE TITLE
test: Add comprehensive integration tests for Regression module [P0]

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#621]
## Overview

This issue tracks the implementation of comprehensive integration tests for the **Regression** module. This is a P0 (Critical Priority) module for supervised learning.

Parent Issue: #615

## Current Status
- **Coverage:** 0% for algorithms (only helpers tested)
- **Source Files:** 45 regression algorithm files
- **Test Files:** Only RegressionHelperIntegrationTests.cs and RegressionDiagnosticsIntegrationTests.cs
- **Note:** Branch `test/regression-integration-tests` may be in progress

## Module Location
`src/Regression/`

### Linear Models
- SimpleRegression
- MultipleRegression
- RidgeRegression
- LassoRegression
- ElasticNetRegression
- OrthogonalRegression
- WeightedRegression

### Regularized/Constrained
- RidgeRegression
- LassoRegression
- ElasticNetRegression
- KernelRidgeRegression

### Tree-Based
- DecisionTreeRegression
- DecisionTreeRegressionBase
- DecisionTreeAsyncRegressionBase
- ConditionalInferenceTreeRegression
- M5ModelTreeRegression
- RandomForestRegression
- ExtremelyRandomizedTreesRegression
- GradientBoostingRegression
- AdaBoostR2Regression

### Non-Linear
- PolynomialRegression
- SplineRegression
- LocallyWeightedRegression (LOESS/LOWESS)
- IsotonicRegression

### Probabilistic
- BayesianRegression
- GaussianProcessRegression
- PoissonRegression
- NegativeBinomialRegression

### Advanced
- PartialLeastSquaresRegression (PLS)
- PrincipalComponentRegression (PCR)
- RadialBasisFunctionRegression
- SupportVectorRegression (SVR)
- QuantileRegression
- QuantileRegressionForests
- RobustRegression

### Specialized
- LogisticRegression
- MultinomialLogisticRegression
- MultivariateRegression
- GeneralizedAdditiveModelRegression (GAM)
- TimeSeriesRegression
- NeuralNetworkRegression
- SymbolicRegression

### Optimization-Based
- GeneticAlgorithmRegression
- StepwiseRegression

### 1. Basic Regression Tests
- [ ] Fit on simple linear data
- [ ] Verify predictions match expected
- [ ] Test coefficient extraction
- [ ] Test intercept calculation

### 2. Algorithm-Specific Tests

**Linear Models:**
- [ ] Test OLS solution matches closed-form
- [ ] Verify R-squared calculation
- [ ] Compare against scikit-learn LinearRegression
- [ ] Test with weighted samples

**Ridge/Lasso/ElasticNet:**
- [ ] Verify regularization effect
- [ ] Test coefficient shrinkage
- [ ] Test sparsity (Lasso)
- [ ] Compare against scikit-learn implementations

**Decision Trees:**
- [ ] Test tree structure
- [ ] Verify split criteria
- [ ] Test pruning behavior
- [ ] Test with different max_depth

**Random Forest/Gradient Boosting:**
- [ ] Test ensemble predictions
- [ ] Verify feature importance
- [ ] Test out-of-bag error (RF)
- [ ] Compare against scikit-learn

**SVR:**
- [ ] Test different kernels (linear, RBF, poly)
- [ ] Verify support vector selection
- [ ] Test epsilon-insensitive zone
- [ ] Compare against scikit-learn SVR

**Gaussian Process:**
- [ ] Test kernel configurations
- [ ] Verify uncertainty estimates
- [ ] Compare against scikit-learn GaussianProcessRegressor

### 3. Prediction Tests
- [ ] Predict on training data
- [ ] Predict on new data
- [ ] Test prediction intervals (where applicable)
- [ ] Test batch prediction

### 4. Metrics and Evaluation
- [ ] R-squared
- [ ] Mean Squared Error (MSE)
- [ ] Root Mean Squared Error (RMSE)
- [ ] Mean Absolute Error (MAE)
- [ ] Residual analysis

### 5. Edge Cases
- [ ] Test with single feature
- [ ] Test with many features (high dimensional)
- [ ] Test with correlated features (multicollinearity)
- [ ] Test with outliers
- [ ] Test with missing values (if supported)
- [ ] Test with constant target
- [ ] Test with zero-variance features

### 6. Serialization Tests
- [ ] Save and load trained models
- [ ] Verify predictions match after reload
- [ ] Test with various configurations

### 7. Clone Tests
- [ ] Clone trained models
- [ ] Verify independence
- [ ] Verify predictions match

### Synthetic Datasets
1. **Perfect linear:** y = 2x + 3 (no noise)
2. **Linear with noise:** y = 2x + 3 + noise
3. **Polynomial:** y = x^2 + 2x + 1
4. **Sinusoidal:** y = sin(x)
5. **Multi-feature:** y = 3*x1 + 2*x2 - x3
6. **High-dimensional:** 100 features

### Real-World Datasets (optional)
- Boston Housing (or California Housing)
- Diabetes dataset

### OLS Solution
```
beta = (X^T X)^(-1) X^T y
```

### Ridge Regression
```
beta = (X^T X + lambda*I)^(-1) X^T y
```

### R-squared
```
R^2 = 1 - SS_res / SS_tot
```

## Priority Order

1. **Critical (Test First):**
   - SimpleRegression
   - MultipleRegression
   - RidgeRegression
   - LassoRegression
   - DecisionTreeRegression
   - RandomForestRegression

2. **High:**
   - ElasticNetRegression
   - GradientBoostingRegression
   - SupportVectorRegression
   - PolynomialRegression
   - LogisticRegression

3. **Medium:**
   - All other regression algorithms

## Acceptance Criteria

- [ ] All 45 regression algorithms have integration tests
- [ ] Basic fit/predict functionality verified
- [ ] Mathematical correctness verified
- [ ] Comparison against scikit-learn where applicable
- [ ] Edge cases tested
- [ ] Serialization works correctly
- [ ] At least 80% code coverage
- [ ] All tests pass on both net8.0 and net471

## References
- scikit-learn regression: https://scikit-learn.org/stable/supervised_learning.html
- statsmodels: https://www.statsmodels.org/stable/regression.html
